lib/scope.js

```diff
@@ -536,7 +535,7 @@ function startScope(basePath, options) {
         _key: key
       , method: method.toUpperCase()
       , basePath: basePath
-      , path: basePathname + uri
+      , path: (typeof uri === 'string') ? basePathname + uri : uri
```

Comment:
tricky point here, because if someone wants to specify part of a `path` at the `basePath` and use regex at the `uri` it won't work. See https://github.com/pgte/nock/blob/master/tests/test_intercept.js#L158-L186 for instance.

Any suggestion?